config/types/iptables.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2017 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"errors"
+	"fmt"
+
+	ignTypes "github.com/coreos/ignition/config/v2_0/types"
+	"github.com/coreos/ignition/config/validate/report"
+	"github.com/vincent-petithory/dataurl"
+)
+
+var (
+	ErrInvalidIptablesDefault = errors.New("iptables chain default must be one of ACCEPT or DROP")
+)
+
+type Iptables struct {
+	V4 *iptablesChains `yaml:"v4"`
+	V6 *iptablesChains `yaml:"v6"`
+}
+
+type iptablesChains struct {
+	Input   iptablesRules `yaml:"input"`
+	Forward iptablesRules `yaml:"forward"`
+	Output  iptablesRules `yaml:"output"`
+}
+
+type iptablesRules struct {
+	Default string   `yaml:"default"`
+	Rules   []string `yaml:"rules"`
+}
+
+func (ir iptablesRules) Validate() report.Report {
+	switch ir.Default {
+	case "ACCEPT", "DROP", "":
+		return report.Report{}
+	default:
+		return report.ReportFromError(ErrInvalidIptablesDefault, report.EntryError)
+	}
+}
+
+func init() {
+	register2_0(func(in Config, out ignTypes.Config) (ignTypes.Config, report.Report) {
+		if in.Iptables != nil {
+			if in.Iptables.V4 != nil {
+				out.Storage.Files = append(out.Storage.Files, ignTypes.File{
+					Filesystem: "root",
+					Path:       "/var/lib/iptables/rules-save",
+					Mode:       0420,
+					Contents:   iptablesContents(in.Iptables.V4),
+				})
+				out.Systemd.Units = append(out.Systemd.Units, ignTypes.SystemdUnit{
+					Name:   "iptables-restore.service",
+					Enable: true,
+				})
+				out.Systemd.Units = append(out.Systemd.Units, ignTypes.SystemdUnit{
+					Name:   "iptables-store.service",
+					Enable: true,
+				})
+			}
+			if in.Iptables.V6 != nil {
+				out.Storage.Files = append(out.Storage.Files, ignTypes.File{
+					Filesystem: "root",
+					Path:       "/var/lib/ip6tables/rules-save",
+					Mode:       0420,
+					Contents:   iptablesContents(in.Iptables.V6),
+				})
+				out.Systemd.Units = append(out.Systemd.Units, ignTypes.SystemdUnit{
+					Name:   "ip6tables-restore.service",
+					Enable: true,
+				})
+				out.Systemd.Units = append(out.Systemd.Units, ignTypes.SystemdUnit{
+					Name:   "ip6tables-store.service",
+					Enable: true,
+				})
+			}
+		}
+		return out, report.Report{}
+	})
+}
+
+func iptablesContents(chains *iptablesChains) ignTypes.FileContents {
+	contents := "*filter\n"
```

Comment:
We should either specify in the docs that this section only configures the filter table or make this configurable. I'm in favor of the latter, since it might prevent needing to go in and change the spec later to accommodate it.